shared/origami/CMakeLists.txt

```
@@ -0,0 +1,156 @@
+# Copyright Advanced Micro Devices, Inc., or its affiliates.
+# SPDX-License-Identifier:  MIT
+
+cmake_minimum_required(VERSION 3.24.4)
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    project(Origami VERSION 1.0.0 LANGUAGES CXX)
+
+    set(ORIGAMI_STANDALONE ON)
+else()
+    set(ORIGAMI_STANDALONE OFF)
+endif()
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(GenerateExportHeader)
+include(CMakeDependentOption)
+
+include(dependencies)
+
+rocm_setup_version(VERSION ${PROJECT_VERSION})
+
+option(ORIGAMI_BUILD_SHARED_LIBS "Build shared libraries." ON)
```

We can do the following or we will need to set `ORIGAMI_BUILD_SHARED_LIBS ` to `OFF` before `add_subdirectory` in hipblaslt:

```suggestion
option(ORIGAMI_BUILD_SHARED_LIBS  "Build shared libraries." ${ORIGAMI_STANDALONE})
```